client/Components/Map.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { GoogleMap, LoadScript, Marker } from '@react-google-maps/api';
+import {TICKETMASTERAPIKEY, REACT_APP_GOOGLEAPIKEY} from '../secret'
+import axios from 'axios'
+class Map extends React.Component{
+  constructor(){
+    super()
+    this.state={
+      lat: 0,
+      lon: 0,
+      ticketDataByLocation: []
+    }
+  }
+  async componentDidMount(){
+    navigator.geolocation.getCurrentPosition((position)=>{
+      this.setState({
```

Comment:
i would like to avoid having two `setStates` like back to back like this- just do it all in one setState

Reply:
I believe we need both setStates because we are using two apis where we need the users location first in order to make use of the lat and long to populate ticketmaster api call, and then we take that info and set it to to state for ticketDataByLocation. If there is a better way pls let me know. Thanks.